src/identity-providers/add/DiscoverySettings.tsx

```diff
@@ -118,6 +128,56 @@ const Fields = ({ readOnly }: DiscoverySettingsProps) => {
           )}
         </>
       )}
+      <SwitchField
+        field="config.pkceEnabled"
+        label="pkceEnabled"
+        isReadOnly={readOnly}
+      />
+      {isPkceEnabled === "true" && (
+        <FormGroup
+          className="pf-u-pb-3xl"
+          label={t("pkceMethod")}
+          labelIcon={
+            <HelpItem
+              helpText="identity-providers-help:pkceMethod"
+              fieldLabelId="identity-providers:pkceMethod"
+            />
+          }
+          fieldId="pkceMethod"
+        >
+          <Controller
+            name="config.pkceMethod"
+            defaultValue={pkceMethods[0]}
+            control={control}
+            render={({ onChange, value }) => (
+              <Select
+                toggleId="pkceMethod"
+                required
+                direction="down"
+                onToggle={() => setPkceMethodOpen(!pkceMethodOpen)}
+                onSelect={(_, value) => {
+                  onChange(value as string);
```

Comment:
Instead of type-casting you can call the `toString()` method, which exists both on `string` and `SelectOptionObject`.
```suggestion
                  onChange(value.toString());
```

Reply:
That fails at run time.  "String.prototype.toString requires that 'this' be a String"

Reply:
Very peculiar, but yeah let's leave this then.